src/measure_theory/decomposition/lebesgue.lean

```diff
@@ -0,0 +1,496 @@
+  -- from this, we can show `ξ + ε * E.indicator` is a function in `measurable_le` with
+  -- integral greater than `ξ`
+    have hξε : ξ + E.indicator (λ _, ε) ∈ measurable_le μ ν,
+    { refine ⟨measurable.add hξm (measurable.indicator measurable_const hE₁), λ A hA, _⟩,
+      have : ∫⁻ a in A, (ξ + E.indicator (λ _, ε)) a ∂μ =
+            ∫⁻ a in A ∩ E, ε + ξ a ∂μ + ∫⁻ a in A ∩ Eᶜ, ξ a ∂μ,
+      { rw [lintegral_add measurable_const hξm, add_assoc,
+            ← lintegral_union (hA.inter hE₁) (hA.inter (hE₁.compl))
+              (disjoint.mono (set.inter_subset_right _ _) (set.inter_subset_right _ _)
+              disjoint_compl_right), set.inter_union_compl],
+        simp_rw [pi.add_apply],
+        rw [lintegral_add hξm (measurable.indicator measurable_const hE₁), add_comm],
+        refine congr_fun (congr_arg has_add.add _) _,
+        rw [set_lintegral_const, lintegral_indicator _ hE₁, set_lintegral_const,
+            measure.restrict_apply hE₁, set.inter_comm] },
+      conv_rhs { rw ← set.inter_union_compl A E },
+      rw [this, measure_union _ (hA.inter hE₁) (hA.inter hE₁.compl)],
+      { exact add_le_add (hε₂ A hA)
+          (hξle (A ∩ Eᶜ) (hA.inter hE₁.compl)) },
+      { exact disjoint.mono (set.inter_subset_right _ _) (set.inter_subset_right _ _)
+          disjoint_compl_right } },
+      have : ∫⁻ a, ξ a + E.indicator (λ _, ε) a ∂μ ≤ Sup (measurable_le_eval μ ν),
+      { exact le_Sup ⟨ξ + E.indicator (λ _, ε), hξε, rfl⟩ },
+  -- but this contradics the maximality of `∫⁻ x, ξ x ∂μ`
+      refine not_lt.2 this _,
+      rw [hξ₁, lintegral_add hξm (measurable.indicator (measurable_const) hE₁),
+          lintegral_indicator _ hE₁, set_lintegral_const],
+      refine ennreal.lt_add_right _ (ennreal.mul_pos.2 ⟨ennreal.coe_pos.2 hε₁, hE₂⟩),
+      have := measure_lt_top (μ.with_density ξ) set.univ,
+      rwa [with_density_apply _ measurable_set.univ, measure.restrict_univ] at this },
+  -- since `μ.with_density ξ ≤ ν`, it is clear that `ν = ν₁ + μ.with_density ξ`
+  { rw hν₁, ext1 A hA,
+    rw [measure.coe_add, pi.add_apply, measure.sub_apply hA hle,
+        add_comm, ennreal.add_sub_cancel_of_le (hle A hA)] },
+end
+
+instance {μ ν : measure α} [finite_measure μ] [finite_measure ν] :
+  finite_measure (singular_part μ ν) :=
+begin
+  obtain ⟨-, -, h⟩ := have_lebesgue_decomposition_spec
+    (have_lebesgue_decomposition_of_finite_measure μ ν),
+  refine finite_measure_le μ _,
+  conv_rhs { rw h },
+  exact measure.le_add_right (le_refl _),
+end
+
+instance {μ ν : measure α} [finite_measure μ] [finite_measure ν] :
+  finite_measure (ν.with_density $ radon_nikodym_deriv μ ν) :=
+begin
+  obtain ⟨-, -, h⟩ := have_lebesgue_decomposition_spec
+    (have_lebesgue_decomposition_of_finite_measure μ ν),
+  refine finite_measure_le μ _,
+  conv_rhs { rw h },
+  exact measure.le_add_left (le_refl _),
+end
+
+/-- The Lebesgue decomposition is unique. -/
+theorem have_lebesgue_decomposition_of_finite_measure_unique
```

Comment:
do you really need to be in any specific setting to have uniqueness in the Lebesgue decomposition theorem? I mean, if you know that `μ = s + ν.with_density f` with `f` measurable and `s` singular with respect to `μ` (without any finiteness assumption), can't you show that these are necessarily the singular part and the Radon-Nikodym derivative? If yes, this could also go in the preliminaries.

Reply:
I think we need at least `\mu` to be finite since otherwise, the decompositions are not necessarily finite and we can't transport them to signed measures. We can relax finiteness on `\nu` if we add the assumption `have_lebesgue_decomposion \mu \nu`. Is this something you think is worth having?

Reply:
I was more thinking of a direct proof. If you have `s + t = s' + t'` where both `s` and `s'` are singular with respect to nu while `t` and `t'` are absolutely continuous with respect to nu, you can find a measurable set `u` with full measure for both s and s', and 0 measure for nu (and therefore for t and t'). Restricting the equality to `u`, you get that `s` and `s'` restricted to `u` are equal, and (since it is full-measure for both of them) then `s = s'`. Arguing in the same way on the complement of `u`, you also get `t = t'`.

Reply:
I've now added the uniqueness statement without the finiteness condition :)